docs/android-sdk.md

```diff
@@ -0,0 +1,519 @@
+## Player Events
+
+It might be important in some cases to receive player events and process the data for your needs. In order to receive player events, a LiveryPlayerListener must be registered to the player.
+
+**_LiveryPlayerListener_**
+
+| Name                                                           | Description                                                                                                                                |
+| -------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| `onActiveQualityChanged(LiveryQuality quality)`                | Called when the active quality of the media being played has changed.                                                                      |
+| `onDurationChanged(long duration)`                             | Called when the duration of media has changed. (Not applicable to live streams.)                                                           |
+| `onPlayerError(Exception e)`                                   | Called when the player encounters an error.                                                                                                |
+| `onPlayerStateChanged(LiveryPlayerState playerState)`          | Called when the playback state of the player has changed.                                                                                  |
+| `onProgressChanged(long buffer, long bufferEnd, long latency)` | Called in short intervals while playing, buffer, buffered position and latency values in milliseconds can be actively retrieved from here. |
+| `onQualitiesChanged(List<LiveryQuality> qualities)`            | Called when available qualities of media have changed.                                                                                     |
+| `onPlaybackRateChanged(float rate)`                            | Called when the playback rate of current media has changed.                                                                                |
+| `onRecovered()`                                                | Called when an error is recovered.                                                                                                         |
+| `onTimeUpdate(long currentTime)`                               | Called when the current time of media has changed.                                                                                         |
+| `onVolumeChanged(float volume)`                                | Called when the volume of the player has changed.                                                                                          |
+| `onSourceChanged(String source)`                               | Called when the media source URL of the player has changed.                                                                                |
+
+### Registering Player Event Listener
```

Comment:
This is a really conventional pattern in Android Java isn't it? Does it need to be extensively documented like this?

Reply:
Normally it should be used with setOnClickListener or addOnClickListener, but since the start it was registerListener so I left it as it is. Therefore an extra explanation is needed.